pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -324,7 +325,13 @@ public static CompletableFuture<ByteBuf> lookupTopicAsync(PulsarService pulsarSe
             }
 
         }).exceptionally(ex -> {
-            if (ex instanceof CompletionException && ex.getCause() instanceof IllegalStateException) {
+            if (ex instanceof CompletionException
+                    && ex.getCause() instanceof BrokerServiceException.ServerMetadataException) {
+                log.warn("Failed to lookup {} for topic {} with error {}", clientAppId, topicName.toString(),
+                        ex.getCause().getMessage());
+                lookupfuture.complete(newLookupErrorResponse(ServerError.MetadataError, ex.getMessage(), requestId));
+                return null;
+            } else if (ex instanceof CompletionException && ex.getCause() instanceof IllegalStateException) {
```

Comment:
We will not get `IllegalStateException` here? since we have changed to `ServerMetadataException`